### PR TITLE
Async/await all the things

### PR DESCRIPTION
**Pull request by JasonEtco**
Back with another PR - this time,  switching some `Promise`s code out for `async/await`. The only real benefit is readability, but it does help a good amount.

I've also done some light cleanup in `log.js`, swapping out `var` for `const` and replacing the `for...of` loop with `Array.map`. Since the end goal is an array of formatting strings, the result should be the same with a little less verbosity - let me know how you feel about that change, or if you prefer it the old way!

### CODE REVIEW

**Comment by bnb** [comment]
Tried running locally and getting an error with array.map: 
image

**Comment by JasonEtco** [comment]
Whoops, fixed in 8c9b46e @bnb! Thanks for checking!

File: lib/search.js [hunk 1]
```diff
@@ -14,26 +14,21 @@ const PAGE = 1 // page_number is 1-based index
...
-        return transform(data)
-      }
-    })
+  if (data.total_count > data.items.length) {
+    const pageCount = Math.ceil(data.total_count / PER_PAGE)
+    const lastPageAllowed = pageCount > MAX_PAGE_ALLOWED ? MAX_PAGE_ALLOWED : pageCount
+    // page_number is 1-based index
+    const randomPage = Math.floor(Math.random() * Math.floor(lastPageAllowed)) + 1
+
+    return octokit.search
```
Comment (hemal7735) [inline lib/search.js]: Is it possible to convert this to async/await as well?
I believe this augments the readability to the changes proposed by you.

Comment (JasonEtco) [reply lib/search.js]: Yep no problem 👍 Done in b3f4d15

**Comment by bnb** [comment]
Merged ❤️

**Pull request merged**